storage/gc_queue.go

```diff
@@ -239,10 +232,23 @@ func (gcq *gcQueue) process(now proto.Timestamp, repl *Replica) error {
 		}
 	}
 
+	done := true
 	if len(intents) > 0 {
+		done = false
 		repl.resolveIntents(repl.context(), intents)
 	}
 
+	// Set start and end keys.
+	if len(gcArgs.Keys) > 0 {
+		done = false
+		gcArgs.Key = gcArgs.Keys[0].Key
+		gcArgs.EndKey = gcArgs.Keys[len(gcArgs.Keys)-1].Key.Next()
+	}
+
+	if done {
+		return nil
+	}
+
 	// Send GC request through range.
```

Comment:
the recipient of the GCRequest does two things:
* `MVCCGarbageCollect` which iterates over all keys in the `GCRequest`
* `MVCCPutProto` of the `GCRequest.GCMeta`. This one has `OldestIntentNanos` set if we have `len(intents) > 0`.

Without the `GCRequest`, the `GCMeta` keeps its old value.
At the very least, this is what the test looks for after the call to `gcqueue.process`. Whether that's the right thing to check or not, I can't tell, but it seems to be.